pymc/tests/test_distributions_moments.py

```diff
@@ -764,3 +766,48 @@ def test_moyal_moment(mu, sigma, size, expected):
     with Model() as model:
         Moyal("x", mu=mu, sigma=sigma, size=size)
     assert_moment_is_expected(model, expected)
+
+
+rand1d = np.random.rand(2)
+rand2d = np.random.rand(2, 3)
+
+
+@pytest.mark.parametrize(
+    "nu, mu, cov, size, expected",
+    [
+        (2, np.ones(1), np.eye(1), None, np.ones(1)),
+        (2, rand1d, np.eye(2), None, rand1d),
+        (2, rand1d, np.eye(2), 2, np.full((2, 2), rand1d)),
+        (2, rand2d, np.eye(3), 2, np.full((2, 2, 3), rand2d)),
```

Comment:
This test combination seems to be failing, because the moment is ignoring broadcasting induced by `nu`

```suggestion
        (np.array([3, 4]), np.ones(2), np.eye(2), None, np.ones((2, 2))),
        (2, rand2d, np.eye(3), 2, np.full((2, 2, 3), rand2d)),
```

Reply:
Tried to fix it. I suppose `nu` and `size` are not going well together. Is the following test correct? if not why?

`(np.array([3, 4]), np.ones(2), np.eye(2), 2, np.ones((2, 2, 2))),`

Reply:
@Sayam753 Can you confirm this is the expected behavior with nu broadcasting?

Reply:
I am first time seeing `nu` not being a scaler. And this is not so straight-forward -

In the case -
```
(np.array([3, 4]), np.ones(2), np.eye(2), 2, np.ones((2, 2, 2)))
size = 2
nu = np.array([3, 4])
mean = np.ones(2)
```

Due to the dimensions of `nu`, we form two independent (batched) MvStudentT distribution. Each random sample from this distribution will be of shape (2, 2). First dimension is implied by `nu` and second by `cov.shape[1]`
Then `size` comes into play and we thus have the shape (2, 2, 2).

<hr>

For the case,

```
size = 2
nu = np.array([3, 4])
mean = np.ones((2, 2))
```

The resulting shape would still be (2, 2, 2) not (2, 2, 2, 2). This is because `mean.shape[:-1]` and `nu.shape` broadcasts with each other. This case is not handled with the current implemented of `get_moment` method.

Reply:
Some pseudo code to fix the same would be -
```
event_shape = cov.shape[-1:]
batch_shape = broadcast_shapes(nu.shape, mu.shape[:-1])
output_shape = size + batch_shape + event_shape
at.fill(output_shape, mu)
```

Reply:
Thanks for the suggestions @Sayam753 and @ricardoV94. I was trying these tests but `result` and `expected` are not matching for these tests. 
For example, if test is:
`(np.array([3, 4]), np.ones(2), np.eye(2), 2, np.ones((2, 2, 2)))`
then in `assert_moment_is_expected`:
```python
result = fn(0)["x"]
result.shape = (2,2)
expected.shape = (2,2,2)
```

I suspect that to enable `nu` broadcasting behaviour, something needs to be changed in the distribution because as per the docs, `MvStudentT` is expecting `nu` as a `float`.
```
Parameters
    ----------
    nu: float
        Degrees of freedom, should be a positive scalar.
    Sigma: matrix
        Covariance matrix. Use `cov` in new code.
    mu: array
        Vector of means.
    cov: matrix
        The covariance matrix.
```

Reply:
Thanks for the suggestions @Sayam753 and @ricardoV94. I was trying these tests but `random_draw` and `expected` are not matching for these tests. 
For example, if test is:
`(np.array([3, 4]), np.ones(2), np.eye(2), 2, np.ones((2, 2, 2)))`
then in `assert_moment_is_expected`:
```python
result = fn(0)["x"]
random_draw = model["x"].eval()
result.shape = (2,2,2)
expected.shape = (2,2,2)
random_draw.shape = (2,2)
```

I suspect that to enable `nu` broadcasting behaviour, something needs to be changed in the distribution because as per the docs, `MvStudentT` is expecting `nu` as a `float`.
```
Parameters
    ----------
    nu: float
        Degrees of freedom, should be a positive scalar.
    Sigma: matrix
        Covariance matrix. Use `cov` in new code.
    mu: array
        Vector of means.
    cov: matrix
        The covariance matrix.
```

Reply:
Yes. Also as per `ndim_params` parameter of MvStudentT here -

https://github.com/pymc-devs/pymc/blob/90423a9c7444b2b5ea2b692921d2ea657d1db087/pymc/distributions/multivariate.py#L264

`nu` parameter is expected to be a scaler. But if there any extra dimensions in `nu`, it should imply batch dimensions and `rng_fn` method should work out of the box. I do not know how aesara handles broadcasting logic, so cc @ricardoV94 or @brandonwillard.

Reply:
> it should imply batch dimensions and rng_fn method should work out of the box

Does this mean the `rng_fn` is working out of the box? We can tweak the shape inference if that's not giving the right results. I am not sure, but I think Aesara will not broadcast any parameters before calling `rng_fn`, broadcasting is only done internally when trying to infer the shape of the RV, but I would need to double check.

If neither the `rng_fn` nor the Aesara shape inference are working for vector `nu`, we can also force it to be a scalar in `make_node` and raise otherwise. 

Also we would need to check if the `logp`/`logcdf` behave properly with vector `nu`, so maybe that's the safest course of action for now? We can open an issue to investigate the behavior with vector `nu`

Reply:
> Does this mean the rng_fn is working out of the box?

Yes. As per @patel-zeel comment two messages above, shape from random samples does not match the shape obtained from initial moments. 

> so maybe forcing it to be a scalar is the safest course of action for now?

Agree. @patel-zeel please revert the `nu` broadcasting logic. And lets only deal with `mu` for moments this time.

Reply:
> > Does this mean the rng_fn is working out of the box?
> 
> Yes. As per @patel-zeel comment two messages above, shape from random samples does not match the shape obtained from initial moments.

Just to be clear, I don't think we should limit the MvStudentT just because the `get_moment` is not giving the right shape. We could limit it on the grounds that `rv.eval().shape` disagrees with `rv.shape.eval()` or that our `logp`/`logcdf` do not work properly with vector `nu`.

Reply:
The shape inference does seem to fail:

```python
x = pm.MvStudentT.dist(nu=[3, 4], mu=np.ones(2), cov=np.eye(2), size=2)
tuple(x.shape.eval()) == tuple(x.eval().shape)  # False
```

Reply:
Thanks for the suggestions @ricardoV94 and @Sayam753. I have disabled `nu` broadcasting for now. I hope I am covering everything else with test cases.